File: src/main/java/com/ggardet/reactive/router/PersonRouter.java
package com.ggardet.reactive.router;

import com.ggardet.reactive.handler.PersonHandler;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.server.RouterFunction;
import org.springframework.web.reactive.function.server.ServerResponse;

import static org.springframework.web.reactive.function.server.RequestPredicates.GET;
import static org.springframework.web.reactive.function.server.RouterFunctions.route;

@Configuration
public class PersonRouter {
    @Bean
    public RouterFunction<ServerResponse> personRoutes(PersonHandler personHandler) {
        return route(GET("/"), personHandler::names);
    }
}

File: src/main/kotlin/com/ggardet/reactive/router/PersonRouter.kt
package com.ggardet.reactive.router

import com.ggardet.reactive.handler.PersonHandler
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.server.RequestPredicates.GET
import org.springframework.web.reactive.function.server.RouterFunction
import org.springframework.web.reactive.function.server.RouterFunctions.route
import org.springframework.web.reactive.function.server.ServerRequest
import org.springframework.web.reactive.function.server.ServerResponse

@Configuration
class PersonRouter {
    @Bean
    fun personRoutes(personHandler: PersonHandler): RouterFunction<ServerResponse> =
        route(GET("/")) { serverRequest: ServerRequest ->
            personHandler.renderPersonsAge(serverRequest)
        }
}
